File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-3.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l=0
        r=0
        total = 0
        res = float("infinity")
        for r in range(len(nums)):
            total += nums[r]
            while l<=r and total>=target:
                res = min(res,r-l+1)
                total -= nums[l]
                l+=1
                   
        return res if res != float("infinity") else 0
        

File: Data Structures & Algorithms/combination-target-sum-ii/submission-3.java
class Solution {
    private Set<List<Integer>> res;
    public List<List<Integer>> combinationSum2(int[] candidates, int target) {
        res = new HashSet<>();
        Arrays.sort(candidates);
        List<Integer> curr = new ArrayList<>();
        dfs(candidates,target,0,curr,0);
        return new ArrayList<>(res);
    }
    private void dfs(int[] candidates,int target,int i, List<Integer> curr, int total){
        if (total==target){
            res.add(new ArrayList<>(curr));
            return;
        }
        if (total>target || i>=candidates.length){
            return;
        }
        curr.add(candidates[i]);
        dfs(candidates,target,i+1,curr,total+candidates[i]);
        curr.remove(curr.size()-1);
        dfs(candidates,target,i+1,curr,total);

    }
}
